src/org/zaproxy/zap/extension/httpsinfo/ExtensionHttpsInfo.java

```diff
@@ -80,8 +82,9 @@ public void unload() {
 		unloaded = true;
 
 		while (!dialogues.isEmpty()) {
-			// When disposed the dialogue is removed from the list
-			dialogues.get(0).dispose();
+			for (HttpsInfoDialog dialog : dialogues) {
```

Comment:
This will lead to a `ConcurrentModificationException` when more than one dialogue is still open (since the list is being changed while iterating it), also the while could be dropped if using the for.